SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
 
-			if (IsValid(PlayerController))
-			{
-				ClientOneSpawnedPawn =
-					GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator, FActorSpawnParameters());
-				RegisterAutoDestroyActor(ClientOneSpawnedPawn);
+		if (AssertIsValid(PlayerController, TEXT("PlayerController should be valid")))
+		{
+			ClientOneSpawnedPawn = GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator);
+			RegisterAutoDestroyActor(ClientOneSpawnedPawn);
 
-				ClientOneDefaultPawn = PlayerController->GetPawn();
+			ClientOneDefaultPawn = PlayerController->GetPawn();
+			PlayerController->Possess(ClientOneSpawnedPawn);
 
-				PlayerController->Possess(ClientOneSpawnedPawn);
+			FinishStep();
+		}
+	});
+
+	// Step 1 - All workers check if they have one AReplicatedGASTestActor in the world, and set a reference to it.
+	AddStep(
+		TEXT("DynamicSubobjectsTestAllWorkers"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			TestActor = GetReplicatedTestActor();
+			FinishStep();
+		},
+		StepTimeLimit);
 
+	// Step 2 - Client 1 checks if it has correctly possessed the TestMovementCharacter.
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientCheckPossesion"), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
+			APawn* PlayerCharacter = GetFlowPawn();
+			if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should be valid"))
+				&& PlayerCharacter == GetFlowPlayerController()->AcknowledgedPawn)
+			{
 				FinishStep();
 			}
-		});
-	}
+		},
+		StepTimeLimit);
 
-	{ // Step 1 - All workers check if they have one AReplicatedGASTestActor in the world, and set a reference to it.
-		AddStep(
-			TEXT("DynamicSubobjectsTestAllWorkers"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
-			[this](float DeltaTime) {
-				TArray<AActor*> FoundActors;
-				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+	// Step 3 - The client checks it has the right initial amount of components
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientCheckNumComponents"), FWorkerDefinition::Client(1), nullptr,
+		[this]() {
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
+			FinishStep();
+		},
+		nullptr);
 
-				if (FoundActors.Num() == 1)
-				{
-					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+	// Step 4 - The server adds the new dynamic component
+	AddStep(
+		TEXT("DynamicSubobjectsTestServerAddComponent"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			// server has 1 more component than client (idk why)
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
 
-					if (IsValid(TestActor))
-					{
-						FinishStep();
-					}
-				}
-			},
-			StepTimeLimit);
-	}
+			// add new dynamic component to test actor
```

Comment:
```suggestion
			// Add new dynamic component to test actor
```
Ultra nit: we usually prefer starting comments with a capital letter